scripts/provisioning/setupcmd.py

```diff
@@ -44,65 +43,68 @@ def __init__(self, config: str):
       raise Exception('empty config URL path')
 
     self._url = config
-    self._s3confstore = S3CortxConfStore(self._url)
+    self._provisioner_confstore = S3CortxConfStore(self._url, 'setup_prov_index')
+    self._s3_confkeys_store = S3CortxConfStore(f'yaml://{self.s3_prov_config}', 'setup_s3keys_index')
+
+    # machine_id will be used to read confstore keys
+    with open('/etc/machine-id') as f:
+      self.machine_id = f.read().strip()
+
+    self.cluster_id = self.provisioner_confstore.get_config(
+      self.s3_confkeys_store.get_config('CONFSTORE_CLUSTER_ID_KEY').format(self.machine_id))
 
   @property
   def url(self) -> str:
     return self._url
 
   @property
-  def s3confstore(self) -> str:
-    return self._s3confstore
+  def provisioner_confstore(self) -> str:
+    return self._provisioner_confstore
+
+  @property
+  def s3_confkeys_store(self) -> str:
+    return self._s3_confkeys_store
 
   def read_ldap_credentials(self):
     """Get 'ldapadmin' user name and password from confstore."""
     try:
-      localconfstore = S3CortxConfStore(f'yaml://{self.s3_prov_config}', 'read_ldap_credentialsidx')
+      s3cipher_obj = CortxS3Cipher(None,
+                                False,
+                                0,
+                                self.s3_confkeys_store.get_config(
+                                  'CONFSTORE_OPENLDAP_CONST_KEY'))
 
-      s3cipher_obj = CortxS3Cipher(None, False, 0, localconfstore.get_config('CONFSTORE_OPENLDAP_CONST_KEY'))
       cipher_key = s3cipher_obj.generate_key()
 
-      encrypted_ldapadmin_pass = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_LDAPADMIN_PASSWD_KEY'))
-      self.ldap_passwd = s3cipher_obj.decrypt(cipher_key, encrypted_ldapadmin_pass)
+      self.ldap_user = self.provisioner_confstore.get_config(
+        self.s3_confkeys_store.get_config(
+          'CONFSTORE_LDAPADMIN_USER_KEY'))
 
-      self.ldap_user = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_LDAPADMIN_USER_KEY'))
+      encrypted_ldapadmin_pass = self.provisioner_confstore.get_config(
+        self.s3_confkeys_store.get_config('CONFSTORE_LDAPADMIN_PASSWD_KEY'))
 
-      encrypted_rootdn_pass = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_ROOTDN_PASSWD_KEY'))
+      encrypted_rootdn_pass = self.provisioner_confstore.get_config(
+        self.s3_confkeys_store.get_config('CONFSTORE_ROOTDN_PASSWD_KEY'))
```

Comment:
@property
self.provisioner_confstore.get_config(self.s3_confkeys_store.get_config())

Reply:
let me take this suggestion in next PR.

Reply:
done